posts/2020-09-01-jakarta-microprofile-beta-20.0.0.10.adoc

```diff
@@ -0,0 +1,336 @@
+
+Applications are expected to provide health check procedures by implementing the HealthCheck interface with the @Liveness or @readiness annotations. These are used by Open Liberty to verify the Liveness or Readiness of the application, respectively. Add the logic of your health check in the call() method, and return the HealthCheckResponse object, by using the simple up()/down() methods from the API:
+
+[source, java]
+----
+**Liveness Check**
+@Liveness
+@ApplicationScoped
+public class AppLiveCheck implements HealthCheck {
+...
+    @Override
+     public HealthCheckResponse call() {
+       ...
+       HealthCheckResponse.up("my-liveness-check");
+       ...
+     }
+}
+
+**Readiness Check**
+@Readiness
+@ApplicationScoped
+public class AppReadyCheck implements HealthCheck {
+...
+    @Override
+     public HealthCheckResponse call() {
+       ...
+       HealthCheckResponse.named("my-app-readiness").status(isMyAppReady()).build();
+       ...
+     }
+}
+...
+----
+
+To view the status of each health check, access the either the `http://<hostname>:<port>/health/live` or `http://<hostname>:<port>/health/ready endpoints`.
+
+More information: 
+* link:https://github.com/eclipse/microprofile-health/releases/tag/3.0-RC3[MicroProfile Health Check 3.0 Release Page]
+* link:https://github.com/eclipse/microprofile-health/blob/master/spec/src/main/asciidoc/release_notes.asciidoc[Release notes]
+
+[#opentracing]
+=== MicroProfile OpenTracing 2.0
+
+MicroProfile OpenTracing 2.0 can be used to profile and monitor applications built using microservice architecture.
+
+MicroProfile OpenTracing 2.0 has upgraded the OpenTracing API to version 0.33.0.  This allows the 
+use of tracing backends and their libraries that are built on OpenTracing API 0.33.0.
+
+==== Try it now
+
+Include the following in the server.xml:
+
+[source, xml]
+----
+    <feature>mpOpenTracing-2.0</feature>
+----
+
+Also configure a tracing backend such as Jaeger or Zipkin 
+    For Jaeger, add the following maven dependencies in the application's pom.xml.
+
+[source, xml]
+----
+<dependency>
+    <groupId>io.jaegertracing</groupId>
+    <artifactId>jaeger-client</artifactId>
+    <version>1.2.0</version>
+</dependency>
+<dependency>
+    <groupId>org.slf4j</groupId>
+    <artifactId>slf4j-api</artifactId>
+    <version>1.7.30</version>
+</dependency>
+<dependency>
+    <groupId>org.slf4j</groupId>
+    <artifactId>slf4j-jdk14</artifactId>
+    <version>1.7.30</version>
+</dependency>
+----
+
+You can find out more about about Jaeger settings set up using environment variables by looking 
+at link:https://github.com/jaegertracing/jaeger-client-java/blob/v1.2.0/jaeger-core/README.md[jaeger-client-java readme].
+
+For the `JAEGER_PASSWORD` environment variable, the password can be encoded using the securityUtility command.
+
+Depending on Jaeger’s sampling settings `JAEGER_SAMPLER_TYPE` and `JAEGER_SAMPLER_PARAM`, 
+Jaeger may not report every spans created by the applications.
```

Comment:
"every spans created" reads oddly - can you check it's correct as plural? (it might be but we should check.

Reply:
I'm going to get in touch with the author and make sure it's right.

Reply:
Changed to "span"